weasyl/profile.py

```diff
@@ -788,28 +781,13 @@ def do_manage(my_userid, userid, username=None, full_name=None, catchphrase=None
 
     # Username
     if username is not None:
-        sysname = d.get_sysname(username)
-
-        if not sysname:
-            raise WeasylError("usernameInvalid")
-        elif d.engine.scalar("SELECT EXISTS (SELECT 0 FROM login WHERE login_name = %(name)s)",
-                             name=sysname):
-            raise WeasylError("usernameExists")
-        elif d.engine.scalar("SELECT EXISTS (SELECT 0 FROM useralias WHERE alias_name = %(name)s)",
-                             name=sysname):
-            raise WeasylError("usernameExists")
-        elif d.engine.scalar("SELECT EXISTS (SELECT 0 FROM logincreate WHERE login_name = %(name)s)",
-                             name=sysname):
-            raise WeasylError("usernameExists")
-
-        with d.engine.begin() as db:
-            db.execute(
-                "UPDATE login SET login_name = %(sysname)s WHERE userid = %(user)s",
-                sysname=sysname, user=userid)
-            db.execute(
-                "UPDATE profile SET username = %(username)s WHERE userid = %(user)s",
-                username=username, user=userid)
-        d._get_display_name.invalidate(userid)
+        login.change_username(
```

Comment:
The way this is coded has an interesting effect. If an admin "updates" the username to the same value (e.g., checks the box, doesn't change the contents), this triggers the cooldown for the username change as if the user made the change themselves, and also states that they can release their currently active username.

I can't tell if this is intended to operate in this fashion, but it's certainly an edge case. (And morbid curiosity did lead me to hit ``Release Old Username``; nothing breaks, thankfully.)


![Wzl-usernamechangeglitch](https://user-images.githubusercontent.com/19718911/74787809-18950e00-527e-11ea-8bff-41590c4173cc.PNG)

Reply:
Oh, good point. I originally had code to allow changes that didn’t affect the resulting `login_name` without adding history entries and took it out for being unnecessary complexity, but requiring the username to actually be different at all is very important (for both normal users and admins).